mongodump/options.go

```diff
@@ -57,7 +57,7 @@ func (inputOptions *InputOptions) GetQuery() ([]byte, error) {
 type OutputOptions struct {
 	Out                        string   `long:"out" value-name:"<directory-path>" short:"o" description:"output directory, or '-' for stdout (default: 'dump')"`
 	Gzip                       bool     `long:"gzip" description:"compress archive or collection output with Gzip"`
-	Oplog                      bool     `long:"oplog" description:"use oplog for taking a point-in-time snapshot"`
+	Oplog                      bool     `long:"oplog" description:"for taking a point-in-time snapshot. For sharded clusters, you MUST disable writes first!"`
```

Comment:
I think we can instead say "for taking a point-in-time snapshot on replica sets that are not part of sharded clusters".
We may have to update this later as part of tools compatibility with invisible sharding.

Reply:
The JIRA ticket says:
> I've seen users attempt to use --oplog in multi-shard environments to back up their clusters without pausing writes. We should update the docs for mongodump and also for mongoreplay --oplogReplay so we don't endorse this practice anymore. 

I understood “this practice” to mean doing a backup/restore against a sharded cluster without pausing writes. But it sounds from your suggestion like we just want to forbid use against sharded clusters in any circumstance. Is that the case?

Reply:
It's more because the benefit of --oplog mode is to allow capturing any writes that occur during the backup. With sharded clusters, we recommend pausing writes in any case, so there isn't much point in using --oplog.

Reply:
OK. So should we _fail_ attempts to use `--oplog` against sharded clusters?

Reply:
We currently fail to execute --oplog when run on a mongos, but we don't stop users from running --oplog on individual shards. We could consider doing this for the future, but that would break user workflows, since unfortunately users are doing this today (without pausing writes). For now I'm hoping we can proceed with a docs change. I'm planning some coordinated messaging to guide people away from these workflows for the future.

Reply:
Is it worth warning if the replica set in question is a shard? That way no one’s workflow suffers, but we better maximize the # of users who will see the guidance.

Reply:
IMO it's preferable to keep the change as docs-only to avoid further complexity.